### PR TITLE
pkg/components: add Inspektor Gadget component

### LINKED ISSUE #1063: Add Inspektor Gadget component
We should add an [Inspektor Gadget](https://github.com/kinvolk/inspektor-gadget) component so users can inspect the cluster and use the Traceloop UI when they have [Headlamp](https://github.com/kinvolk/headlamp/) installed.

### PR DESCRIPTION
**Pull request by iaguis**
Fixes #1063

### CODE REVIEW

File: assets/charts/components/inspektor-gadget/templates/psp.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: inspektor-gadget-restricted
```
Comment (invidian) [inline assets/charts/components/inspektor-gadget/templates/psp.yaml:4]: `-restricted`? From what? :joy: This is almost the most permissive policy I've seen :smile: 
```suggestion
  name: inspektor-gadget
```

File: assets/charts/components/inspektor-gadget/values.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Default values for inspektor-gadget.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+image:
+  repository: docker.io/kinvolk/gadget
```
Comment (invidian) [inline assets/charts/components/inspektor-gadget/values.yaml:6]: `docker.io` is default ?
```suggestion
  repository: kinvolk/gadget
```

**Review by invidian — DISMISSED** [review]
Just some nits, otherwise LGTM, but I'm curious why CI is failing.

File: assets/charts/components/inspektor-gadget/templates/psp.yaml (hunk 1 above)
Comment (iaguis) [reply assets/charts/components/inspektor-gadget/templates/psp.yaml:4]: at least it can't use Secrets? :joy:

**Comment by iaguis** [comment]
I've changed the namespace to `gadget`, for some reasoning see https://github.com/kinvolk/inspektor-gadget/pull/140#issuecomment-709934072.

> I'm curious why CI is failing.

I was using `WaitForDeployment()` instead of `ẀaitForDaemonSet()` :facepalm:

**Comment by iaguis** [comment]
> I've changed the namespace to `gadget`, for some reasoning see [kinvolk/inspektor-gadget#140 (comment)](https://github.com/kinvolk/inspektor-gadget/pull/140#issuecomment-709934072).

After thinking about it some more and discussing with others I think what makes most sense is to keep IG in the `kube-system` namespace so it works out of the box with the current `kubectl gadget` plugin. Since it will have its own PSP this should be OK. If/when upstream IG changes its namespace to `gadget` we can change it too.

I changed the PR accordingly.

**Comment by invidian** [comment]
Odd that CI didn't get triggered.

File: docs/configuration-reference/components/inspektor-gadget.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+```
+
+## Attribute reference
+
+Table of all the arguments accepted by the component.
+
+Example:
+
+| Argument           | Description                                                                                                                   | Default       | Type   | Required |
+|--------------------|-------------------------------------------------------------------------------------------------------------------------------|--:-:----------|--:-:---|--:-:-----|
```
Comment (invidian) [inline docs/configuration-reference/components/inspektor-gadget.md:50]: `:-:` does not render properly on GitHub.

**Review by invidian — DISMISSED** [review]
Just one broken thing, otherwise LGTM.

File: assets/charts/components/inspektor-gadget/values.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Default values for inspektor-gadget.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+image:
+  repository: kinvolk/gadget
```
Comment (knrt10) [inline assets/charts/components/inspektor-gadget/values.yaml:6]: Unrelated but we should also push it to quay

File: test/components/inspektor-gadget/inspektorgadget_test.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+	testutil "github.com/kinvolk/lokomotive/test/components/util"
+)
+
+const (
+	daemonset = "inspektor-gadget"
+	namespace = "kube-system"
+)
+
+func TestInspektorGadgetDeployment(t *testing.T) {
```
Comment (knrt10) [inline test/components/inspektor-gadget/inspektorgadget_test.go:31]: TestInspektorGadgetDaemonSet

**Review by knrt10 — CHANGES_REQUESTED** [review]
One small NIT, rest LGTM

File: assets/charts/components/inspektor-gadget/values.yaml (hunk 4 above)
Comment (iaguis) [reply assets/charts/components/inspektor-gadget/values.yaml:6]: Filed an issue upstream: https://github.com/kinvolk/inspektor-gadget/issues/141

Comment (knrt10) [reply assets/charts/components/inspektor-gadget/values.yaml:6]: cool

**Review by knrt10 — APPROVED** [review]
Looks great. +1

**Comment by iaguis** [comment]
I'll take @invidian's lgtm and @knrt10's one and merge this.

**Pull request merged**